retail/interactive-tutorial/RetailProducts.Samples/CreateProductSample.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_create_product]
+// Create product in a catalog using Retail API
+
+using Google.Cloud.Retail.V2;
+using System;
+
+/// <summary>
+/// The create product sample class.
+/// </summary>
+public class CreateProductSample
+{
+    /// <summary>
+    /// Generate product with fulfillment info.
+    /// </summary>
+    /// <returns>Generated product.</returns>
+    private static Product GenerateProductWithFulfillment()
+    {
+        Product generatedProduct = new Product
+        {
+            Id = Guid.NewGuid().ToString("N").Substring(0, 14),
+            Title = "Nest Mini",
+            Type = Product.Types.Type.Primary,
+            PriceInfo = new PriceInfo
+            {
+                Price = 30.0f,
+                OriginalPrice = 35.5f,
+                CurrencyCode = "USD"
+            },
+            Availability = Product.Types.Availability.InStock,
+            Categories = { "Speakers and displays" },
+            Brands = { "Google" }
+        };
+
+        string[] placeIds = { "store0", "store1" };
+
+        FulfillmentInfo fulfillmentInfo = new FulfillmentInfo
+        {
+            Type = "pickup-in-store"
+        };
+
+        fulfillmentInfo.PlaceIds.AddRange(placeIds);
+
+        generatedProduct.FulfillmentInfo.Add(fulfillmentInfo);
+
+        return generatedProduct;
+    }
+
+    /// <summary>
+    /// Get create product request.
+    /// </summary>
+    /// <param name="productToCreate">The actual product object to create.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>Create product request.</returns>
+    private static CreateProductRequest GetCreateProductRequest(Product productToCreate, string projectId)
+    {
+        string defaultBranchName = $"projects/{projectId}/locations/global/catalogs/default_catalog/branches/default_branch";
```

Comment:
One other example of what I mentioned on the other PR about there being a type `BranchName` or similar that you can use to create these resource names.

Reply:
I will refactor the way we are creating the Branches/Catalogs in a separate PR by your suggestion.